esx_service/vmdk_ops.py

```diff
@@ -838,25 +841,25 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     except ValidationError as ex:
         return err(str(ex))
 
+    if not datastore:
+        datastore = default_datastore 
+
     if datastore and not vmdk_utils.validate_datastore(datastore):
         return err("Invalid datastore '%s'.\n" \
                    "Known datastores: %s.\n" \
                    "Default datastore: %s" \
                    % (datastore, ", ".join(get_datastore_names_list()), default_datastore))
 
-    if not datastore:
-        datastore_url = default_datastore_url
-        datastore = default_datastore
-    else:
-        datastore_url = vmdk_utils.get_datastore_url(datastore)
+    # datastore is present and validated.
+    datastore_url = vmdk_utils.get_datastore_url(datastore)
+
+    logging.debug("executeRequest: vm_uuid=%s, vm_name=%s, tenant_name=%s, tenant_uuid=%s, default_datastore_url=%s",
+                  vm_uuid, vm_name, tenant_uuid, tenant_name, default_datastore_url)
 
     error_info, tenant_uuid, tenant_name = auth.authorize(vm_uuid, datastore_url, cmd, opts)
     if error_info:
         return err(error_info)
 
-    # get /vmfs/volumes/<volid>/dockvols path on ESX:
-    datastore = get_datastore_name(datastore_url)
-
```

Comment:
The current code has the same logic distributed in the first part of executeRequest. I've consolidated the datastore and datastore_url selection above.